### DISCUSSION TITLE
csharp-ls error

### DISCUSSION THREAD

**Discussion by henrybley** [nix-community/nixvim#482] [Q&A] [answered]
Hi, I am new to nix and vim and trying to use nixvim but I am getting this error when trying to rebuild.

`error: attribute 'csharp-ls' missing

       at /nix/store/9iswkvfj5yrnjc0np1xvgxmyw3bsfsss-source/plugins/lsp/language-servers/default.nix:45:17:

           44|       description = "Enable csharp-ls, for C#.";
           45|       package = pkgs.csharp-ls;
             |                 ^
           46|       serverName = "csharp_ls";`

**Comment by traxys** [comment, marked as answer]
You need to have a quite up to date nixpkgs-unstable

**Reply by henrybley** [reply]
thanks, that solved it for me

**Reply by arunoruto** [reply]
So people have to use the unstable branch until csharp-ls is available in stable (like 23.11)?

**Reply by siph** [reply]
I'm pretty sure you're supposed to match your `nixvim` and `nixpkgs` branches. If you're on `nixos-unstable`/`nixpkgs-unstable` you use the `main` branch. If you're on `nixos-23.05` you use the `nixvim/nixos-23.05` branch.

**Reply by traxys** [reply]
Exactly, tough there still could be *some* breakage as we were not as carefull as could be before the split. But we only test nixvim with the unstable input of the flake, and it is never older than 1 week.

**Reply by arunoruto** [reply]
Thanks for the reply! And it seems like I got it somehow :) I am still new to nix, so I'm learning along.
I will put here what I did to make it work, if someone else got the same question:
```nix
{ pkgs, lib, ... }:
let
  nixvim = import (builtins.fetchGit {
    url = "https://github.com/nix-community/nixvim";
    ref = "nixos-23.05"; # Change this with newer versions
  });
in
{
  imports = [ nixvim.homeManagerModules.nixvim ];
  ...
}
```
